features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/net/BatchesToSegmentsMapper.kt

```diff
@@ -24,14 +24,14 @@ import com.google.gson.JsonParser
  */
 internal class BatchesToSegmentsMapper(private val internalLogger: InternalLogger) {
 
-    fun map(batchData: List<ByteArray>): Pair<MobileSegment, JsonObject>? {
+    fun map(batchData: List<ByteArray>): List<Pair<MobileSegment, JsonObject>> {
         return groupBatchDataIntoSegments(batchData)
     }
 
     // region Internal
 
-    private fun groupBatchDataIntoSegments(batchData: List<ByteArray>): Pair<MobileSegment, JsonObject>? {
-        val reducedEnrichedRecord = batchData
+    private fun groupBatchDataIntoSegments(batchData: List<ByteArray>): List<Pair<MobileSegment, JsonObject>> {
+        return batchData
             .asSequence()
```

Comment:
btw, what is the benefit of using `asSequence`? We will still iterate over the returned collection, so all the operators will be run anyway.

Reply:
The doc for `Sequence` says:

```
 * A sequence that returns values through its iterator. The values are evaluated lazily, and the sequence
 * is potentially infinite.
```

We have a finite collection.

Reply:
yes but in case one of the object is broken in one of the lower layers it will be removed from the lower iterations...it's debatable. I would still keep it though.

Reply:
What do you mean? That is the case already with the normal Kotlin collection operations - they are applied on top of the previous ones.

Reply:
you have a more in detail explanation here, but usually it makes sense to use `sequence` when you have more operators: https://typealias.com/guides/when-to-use-sequences/